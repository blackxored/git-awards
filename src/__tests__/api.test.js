import { back } from 'nock';
import {
  getUser,
  getTopUsers,
  getRankForLanguage,
  getUserBestRankWorldwide,
} from '../api';

back.fixtures = `${__dirname}/__fixtures__/`;
back.setMode('lockdown');

describe('api', () => {
  describe('.getUser', () => {
    it('gets a user by username', async done => {
      back('blackxored.json', async nockDone => {
        const user = await getUser('blackxored');
        expect(user).toMatchSnapshot();

        nockDone();
        done();
      });
    });
  });

  describe('.getRankForLanguage', () => {
    it('returns a specific rank for a language', done => {
      back('blackxored.json', async nockDone => {
        const result = getRankForLanguage(
          'javascript',
          await getUser('blackxored'),
        );
        expect(result).toMatchSnapshot();

        nockDone();
        done();
      });
    });
  });

  describe('.getUserBestRankWorldwide', () => {
    it('returns the rank where the user is best-ranked worlwide', done => {
      back('blackxored.json', async nockDone => {
        const result = getUserBestRankWorldwide(await getUser('blackxored'));

        expect(result.language).toEqual('ruby');

        nockDone();
        done();
      });
    });
  });

  describe('.getTopUsers', () => {
    it('returns the list of top users', done => {
      back('top-users.json', async nockDone => {
        const result = await getTopUsers();
        expect(result).toMatchSnapshot();

        expect(result[0].login).toEqual('freecodecamp');

        nockDone();
        done();
      });
    });
  });
});
