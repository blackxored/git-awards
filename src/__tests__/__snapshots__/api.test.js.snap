// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api .getRankForLanguage returns a specific rank for a language 1`] = `
Object {
  "cityCount": 5,
  "cityRank": null,
  "countryCount": 5,
  "countryRank": null,
  "language": "javascript",
  "repositoryCount": 34,
  "starsCount": 13,
  "worldCount": 889848,
  "worldRank": 67844,
}
`;

exports[`api .getTopUsers returns the list of top users 1`] = `
Array [
  Object {
    "city": "san francisco",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 1,
    "gravatar_url": "https://avatars.githubusercontent.com/u/9892522?v=3",
    "id": 14354665,
    "login": "freecodecamp",
    "stars_count": 296762,
    "world_rank": 1,
  },
  Object {
    "city": "menlo park",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 2,
    "gravatar_url": "https://avatars.githubusercontent.com/u/69631?v=3",
    "id": 5156961,
    "login": "facebook",
    "stars_count": 269189,
    "world_rank": 2,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/6128107?v=3",
    "id": 10673069,
    "login": "vuejs",
    "stars_count": 130710,
    "world_rank": 3,
  },
  Object {
    "city": "menlo park",
    "city_rank": 2,
    "country": "united states",
    "country_rank": 3,
    "gravatar_url": "https://avatars.githubusercontent.com/u/6412038?v=3",
    "id": 10949229,
    "login": "reactjs",
    "stars_count": 113431,
    "world_rank": 4,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/139426?v=3",
    "id": 5222943,
    "login": "angular",
    "stars_count": 113057,
    "world_rank": 5,
  },
  Object {
    "city": "san francisco",
    "city_rank": 2,
    "country": "united states",
    "country_rank": 4,
    "gravatar_url": "https://avatars.githubusercontent.com/u/698437?v=3",
    "id": 5742137,
    "login": "airbnb",
    "stars_count": 107937,
    "world_rank": 6,
  },
  Object {
    "city": "mountain view",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 5,
    "gravatar_url": "https://avatars.githubusercontent.com/u/131524?v=3",
    "id": 5215536,
    "login": "mozilla",
    "stars_count": 90046,
    "world_rank": 7,
  },
  Object {
    "city": "boston",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 6,
    "gravatar_url": "https://avatars.githubusercontent.com/u/70142?v=3",
    "id": 5157444,
    "login": "jquery",
    "stars_count": 89175,
    "world_rank": 8,
  },
  Object {
    "city": "oakland",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 7,
    "gravatar_url": "https://avatars.githubusercontent.com/u/12631?v=3",
    "id": 5103040,
    "login": "substack",
    "stars_count": 83392,
    "world_rank": 9,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/1562726?v=3",
    "id": 6553218,
    "login": "d3",
    "stars_count": 79892,
    "world_rank": 10,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/1715673?v=3",
    "id": 6692458,
    "login": "zeit",
    "stars_count": 77755,
    "world_rank": 11,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/170270?v=3",
    "id": 5250753,
    "login": "sindresorhus",
    "stars_count": 75492,
    "world_rank": 12,
  },
  Object {
    "city": "san antonio",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 8,
    "gravatar_url": "https://avatars.githubusercontent.com/u/887802?v=3",
    "id": 5921764,
    "login": "automattic",
    "stars_count": 69494,
    "world_rank": 13,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/1778935?v=3",
    "id": 6749792,
    "login": "googlechrome",
    "stars_count": 66700,
    "world_rank": 14,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/319357?v=3",
    "id": 5384787,
    "login": "nodejs",
    "stars_count": 61661,
    "world_rank": 15,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/5658226?v=3",
    "id": 10215897,
    "login": "expressjs",
    "stars_count": 59770,
    "world_rank": 16,
  },
  Object {
    "city": "san francisco",
    "city_rank": 3,
    "country": "united states",
    "country_rank": 9,
    "gravatar_url": "https://avatars.githubusercontent.com/u/230541?v=3",
    "id": 5307300,
    "login": "mbostock",
    "stars_count": 57708,
    "world_rank": 17,
  },
  Object {
    "city": "victoria",
    "city_rank": 1,
    "country": "canada",
    "country_rank": 2,
    "gravatar_url": "https://avatars.githubusercontent.com/u/25254?v=3",
    "id": 5115044,
    "login": "tj",
    "stars_count": 57693,
    "world_rank": 18,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/1342004?v=3",
    "id": 6350341,
    "login": "google",
    "stars_count": 54504,
    "world_rank": 19,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/2105791?v=3",
    "id": 7041548,
    "login": "webpack",
    "stars_count": 53291,
    "world_rank": 20,
  },
  Object {
    "city": "moscow",
    "city_rank": 1,
    "country": "russia",
    "country_rank": 1,
    "gravatar_url": "https://avatars.githubusercontent.com/u/810438?v=3",
    "id": 5848236,
    "login": "gaearon",
    "stars_count": 52360,
    "world_rank": 21,
  },
  Object {
    "city": "seattle",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 10,
    "gravatar_url": "https://avatars.githubusercontent.com/u/5078602?v=3",
    "id": 9651367,
    "login": "formidablelabs",
    "stars_count": 51533,
    "world_rank": 22,
  },
  Object {
    "city": "philadelphia",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 11,
    "gravatar_url": "https://avatars2.githubusercontent.com/u/5502029?v=4",
    "id": 10063722,
    "login": "typicode",
    "stars_count": 50515,
    "world_rank": 23,
  },
  Object {
    "city": "new york",
    "city_rank": 1,
    "country": "united states",
    "country_rank": 12,
    "gravatar_url": "https://avatars.githubusercontent.com/u/4732?v=3",
    "id": 5095505,
    "login": "jashkenas",
    "stars_count": 49542,
    "world_rank": 24,
  },
  Object {
    "city": null,
    "city_rank": null,
    "country": null,
    "country_rank": null,
    "gravatar_url": "https://avatars.githubusercontent.com/u/10566080?v=3",
    "id": 15018144,
    "login": "socketio",
    "stars_count": 48439,
    "world_rank": 25,
  },
]
`;

exports[`api .getUser gets a user by username 1`] = `
Object {
  "city": null,
  "country": null,
  "gravatarUrl": "https://avatars3.githubusercontent.com/u/133308?v=4",
  "id": 5217195,
  "login": "blackxored",
  "rankings": Array [
    Object {
      "cityCount": 0,
      "cityRank": null,
      "countryCount": 0,
      "countryRank": null,
      "language": "ruby",
      "repositoryCount": 44,
      "starsCount": 16,
      "worldCount": 317144,
      "worldRank": 16465,
    },
    Object {
      "cityCount": 5,
      "cityRank": null,
      "countryCount": 5,
      "countryRank": null,
      "language": "javascript",
      "repositoryCount": 34,
      "starsCount": 13,
      "worldCount": 889848,
      "worldRank": 67844,
    },
    Object {
      "cityCount": 0,
      "cityRank": null,
      "countryCount": 0,
      "countryRank": null,
      "language": "go",
      "repositoryCount": 1,
      "starsCount": 1,
      "worldCount": 68779,
      "worldRank": 35002,
    },
    Object {
      "cityCount": 2,
      "cityRank": null,
      "countryCount": 2,
      "countryRank": null,
      "language": "html",
      "repositoryCount": 1,
      "starsCount": 0,
      "worldCount": 264426,
      "worldRank": 187698,
    },
    Object {
      "cityCount": 0,
      "cityRank": null,
      "countryCount": 0,
      "countryRank": null,
      "language": "perl",
      "repositoryCount": 1,
      "starsCount": 0,
      "worldCount": 42885,
      "worldRank": 34480,
    },
    Object {
      "cityCount": 5,
      "cityRank": null,
      "countryCount": 5,
      "countryRank": null,
      "language": "python",
      "repositoryCount": 1,
      "starsCount": 0,
      "worldCount": 511149,
      "worldRank": 422264,
    },
    Object {
      "cityCount": 0,
      "cityRank": null,
      "countryCount": 0,
      "countryRank": null,
      "language": "c#",
      "repositoryCount": 1,
      "starsCount": 0,
      "worldCount": 201959,
      "worldRank": 158450,
    },
    Object {
      "cityCount": 2,
      "cityRank": null,
      "countryCount": 2,
      "countryRank": null,
      "language": "shell",
      "repositoryCount": 5,
      "starsCount": 0,
      "worldCount": 215011,
      "worldRank": 108255,
    },
    Object {
      "cityCount": 1,
      "cityRank": null,
      "countryCount": 1,
      "countryRank": null,
      "language": "coffeescript",
      "repositoryCount": 1,
      "starsCount": 0,
      "worldCount": 44080,
      "worldRank": 34253,
    },
    Object {
      "cityCount": 3,
      "cityRank": null,
      "countryCount": 3,
      "countryRank": null,
      "language": "css",
      "repositoryCount": 3,
      "starsCount": 0,
      "worldCount": 417253,
      "worldRank": 138483,
    },
  ],
}
`;
